gammapy/data/data_store.py

```diff
@@ -226,14 +229,17 @@ def obs(self, obs_id):
         kwargs = {"obs_id": int(obs_id)}
         kwargs["obs_info"] = table_row_to_dict(row)
 
-        hdu_list = ["events", "gti", "aeff", "edisp", "psf", "bkg", "rad_max"]
+        if required_hdu is None or required_hdu==[]:
+            hdu_list = ["events", "gti", "aeff", "edisp", "psf", "bkg", "rad_max"]
```

Comment:
So if the `required_hdu` is `None`, is tries to load *all* of them? And raises warning for the missing ones?

Reply:
That's very contra-intuitive. Requiring `None` or an empty list should not be treated as `all`, since it is semantically the exact opposite.

Reply:
@AtreyeeS yes
@maxnoe I interpret None required as all optional

Reply:
I think having an info table will indeed be very helpful. It seems that in the default case, the user will end up with a bunch of warnings (eg: either a `rad_max` or `psf` warning will be there for *all*) and then if you have say, an `edisp` missing somewhere, it becomes impossible to find. And `MapMaker` breaks later on.

Maybe @maxnoe suggestion of having `full_enclosure` as the default is better?